core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -215,8 +216,15 @@ public TableBuilder withProperty(String key, String value) {
     @Override
     public Table create() {
       TableOperations ops = newTableOps(identifier);
-      if (ops.current() != null) {
-        throw new AlreadyExistsException("Table already exists: %s", identifier);
+      try {
+        if (ops.current() != null) {
+          throw new AlreadyExistsException("Table already exists: %s", identifier);
+        }
+      } catch (NoSuchIcebergTableException ne) {
```

Comment:
Is this correct?

If this happens, then the table exists and is not an Iceberg table. That is expected if we are creating the table from the Hive hook, but in that case I think that we will have injected metadata in the hook that signals to the `TableOperations` implementation that the table was pre-created by Hive and should be replaced (so update will be used for the initial commit). If we detect that in `TableOperations` then we should be able to avoid throwing this exception and return null like before.

If the operation is not happening inside of the Hive hooks, then we do want to let the exception propagate because there is an existing Hive table.

Reply:
Removed. With the new HMS table property this is not needed anymore.